packages/common/infra/src/page/record-list.ts

```diff
@@ -31,7 +30,12 @@ export class PageRecordList {
       return () => {
         dispose();
       };
-    }),
+    }).pipe(
+      distinctUntilChanged((p, c) => isEqual(p, c)),
+      map(ids =>
+        ids.map(id => new PageRecord(id, this.workspace, this.localState))
```

Comment:
Any idea on why recreating `PageRecord` is resource-intensive?

Reply:
Because our pagelist is too long, there are 500+